### PR TITLE
Add fluid dex connector

### PR DESCRIPTION
**Pull request by shriyatyagii**

### CODE REVIEW

File: contracts/mainnet/connectors/fluid-dex/main.sol [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+            token1Amt_,
+            minSharesAmt_,
+            estimate_
+        );
+
+        _eventName = "LogFluidDexDeposit(address,uint256,uint256,uint256,bool)";
+        _eventParam = abi.encode(
+            dex_,
+            constantsView_.token0,
+            constantsView_.token1,
```
Comment (thrilok209) [inline contracts/mainnet/connectors/fluid-dex/main.sol:84]: @shriyatyagii No need for these

File: contracts/mainnet/connectors/fluid-dex/main.sol [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+     * @param token1Amt_ The amount of token1 to deposit
+     * @param minSharesAmt_ The minimum amount of shares the user expects to receive
+     * @param estimate_ If true, function will revert with estimated shares without executing the deposit
+     */
+    function depositInDex(
+        address dex_,
+        uint256 token0Amt_,
+        uint256 token1Amt_,
+        uint256 minSharesAmt_,
+        bool estimate_
```
Comment (thrilok209) [inline contracts/mainnet/connectors/fluid-dex/main.sol:35]: @shriyatyagii We can remove this and pass always true

File: contracts/mainnet/connectors/fluid-dex/main.sol [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+            approve(TokenInterface(constantsView_.token1), dex_, token1Amt_);
+        }
+
+        // Deposit in Fluid Dex
+        IFluidDex(dex_).deposit{value: ethAmt_}(
+            token0Amt_,
+            token1Amt_,
+            minSharesAmt_,
+            estimate_
```
Comment (thrilok209) [inline contracts/mainnet/connectors/fluid-dex/main.sol:77]:
```suggestion
            true
```

File: contracts/mainnet/connectors/fluid-dex/main.sol [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+        _eventName = "LogFluidDexDeposit(address,uint256,uint256,uint256,bool)";
+        _eventParam = abi.encode(
+            dex_,
+            constantsView_.token0,
+            constantsView_.token1,
+            token0Amt_,
+            token1Amt_,
+            minSharesAmt_,
+            estimate_
```
Comment (thrilok209) [inline contracts/mainnet/connectors/fluid-dex/main.sol:88]: remove this as well

**Comment by shriyatyagii** [comment]
Add connector -
`FLUID-DEX-A`: [0x7FF4C3F6CA8661C072A103c6491ef20Edb9b66db](https://etherscan.io/address/0x7FF4C3F6CA8661C072A103c6491ef20Edb9b66db#code)

File: contracts/mainnet/connectors/fluid-dex/main.sol (hunk 4 above)
Comment (shriyatyagii) [reply contracts/mainnet/connectors/fluid-dex/main.sol]: Removed

File: contracts/mainnet/connectors/fluid-dex/main.sol (hunk 2 above)
Comment (shriyatyagii) [reply contracts/mainnet/connectors/fluid-dex/main.sol]: Removed

File: contracts/mainnet/connectors/fluid-dex/main.sol (hunk 1 above)
Comment (shriyatyagii) [reply contracts/mainnet/connectors/fluid-dex/main.sol]: Removed

**Pull request merged**